src/sage/combinat/designs/block_design.py

```diff
@@ -318,15 +318,15 @@ def DesarguesianProjectivePlaneDesign(n, point_coordinates=True, check=True):
 
     EXAMPLES::
 
-        sage: designs.DesarguesianProjectivePlaneDesign(2)
+        sage: designs.DesarguesianProjectivePlaneDesign(2)                              # optional - sage.rings.finite_rings
```

Comment:
Here we could put `sage.rings.finite_rings` at the top of the file as it seems mandatory for a large portion of the tests.

Reply:
Indeed looking at 
```
for a in src/sage/combinat/designs/*.py src/sage/combinat/designs/*.pyx; do echo "$a: $(grep -c '# optional' $a) optional / $(grep -c 'sage: ' $a) doctests" ; done                   git:sage_combinat_more_optional
src/sage/combinat/designs/all.py: 0 optional / 0 doctests
src/sage/combinat/designs/bibd.py: 8 optional / 141 doctests
src/sage/combinat/designs/block_design.py: 69 optional / 116 doctests
src/sage/combinat/designs/covering_design.py: 3 optional / 49 doctests
src/sage/combinat/designs/database.py: 97 optional / 395 doctests
src/sage/combinat/designs/design_catalog.py: 4 optional / 5 doctests
src/sage/combinat/designs/difference_family.py: 121 optional / 362 doctests
src/sage/combinat/designs/difference_matrices.py: 0 optional / 20 doctests
src/sage/combinat/designs/ext_rep.py: 2 optional / 102 doctests
src/sage/combinat/designs/group_divisible_designs.py: 0 optional / 30 doctests
src/sage/combinat/designs/incidence_structures.py: 18 optional / 350 doctests
src/sage/combinat/designs/latin_squares.py: 0 optional / 43 doctests
src/sage/combinat/designs/orthogonal_arrays.py: 0 optional / 187 doctests
src/sage/combinat/designs/orthogonal_arrays_build_recursive.py: 0 optional / 73 doctests
src/sage/combinat/designs/resolvable_bibd.py: 0 optional / 22 doctests
src/sage/combinat/designs/steiner_quadruple_systems.py: 0 optional / 37 doctests
src/sage/combinat/designs/twographs.py: 20 optional / 35 doctests
src/sage/combinat/designs/designs_pyx.pyx: 34 optional / 99 doctests
src/sage/combinat/designs/evenly_distributed_sets.pyx: 7 optional / 50 doctests
src/sage/combinat/designs/gen_quadrangles_with_spread.pyx: 0 optional / 52 doctests
src/sage/combinat/designs/orthogonal_arrays_find_recursive.pyx: 0 optional / 61 doctests
src/sage/combinat/designs/subhypergraph_search.pyx: 0 optional / 12 doctests
```
... this file is worth a look.

Reply:
OK, good call. Done in 62d3d48d42 (+ cosmetics)